rust/src/push/evaluator.rs

```diff
@@ -117,7 +117,7 @@ impl PushRuleEvaluator {
         msc3931_enabled: bool,
     ) -> Result<Self, Error> {
         let body = match flattened_keys.get("content.body") {
-            Some(JsonValue::Value(SimpleJsonValue::Str(s))) => s.clone(),
+            Some(JsonValue::Value(SimpleJsonValue::Str(s))) => s.clone().to_string(),
```

Comment:
```suggestion
            Some(JsonValue::Value(SimpleJsonValue::Str(s))) => s.clone().into_owned(),
```

is probably more technically efficient (reduces it to one clone). The alternative would also be `s.to_string()`.